core/src/impl/Kokkos_ViewArray.hpp

```diff
@@ -350,7 +350,8 @@ class ViewMapping<Traits, Kokkos::Array<>> {
   template <class... P>
   Kokkos::Impl::SharedAllocationRecord<> *allocate_shared(
       Kokkos::Impl::ViewCtorProp<P...> const &arg_prop,
-      typename Traits::array_layout const &arg_layout) {
+      typename Traits::array_layout const &arg_layout,
+      bool execution_space_specified) {
```

Comment:
Can't you deduce it from `arg_prop`?

Reply:
No, at this point `arg_prop` always has an execution space. Before calling this function we explicitly add it if the original didn't have an execution space.